src/main/java/com/powsybl/openloadflow/sensi/DcSensitivityAnalysis.java

```diff
@@ -617,52 +694,105 @@ public void checkContingencies(LfNetwork lfNetwork, List<Contingency> contingenc
-                    value = 0d;
+                    sensiValue = 0d;
+                    flowValue = 0d;
                     break;
                 }
-                value = value + contingencyElement.getAlpha() * p1.calculate(contingenciesStates, contingencyElement.getContingencyIndex());
+                double contingencySensitivity = p1.calculate(contingenciesStates, contingencyElement.getContingencyIndex());
+                flowValue += contingencyElement.getAlphaFlow() * contingencySensitivity;
+                sensiValue +=  contingencyElement.getAlphaSensi() * contingencySensitivity;
+            }
+        }
+        return new SensitivityValue(factor.getFactor(), sensiValue * PerUnit.SB, flowValue * PerUnit.SB, 0);
+    }
+
+    protected void setBaseSensitivityValues(List<SensitivityFactorGroup> factorGroups, DenseMatrix factorsState) {
+        for (SensitivityFactorGroup factorGroup : factorGroups) {
+            for (LfSensitivityFactor factor : factorGroup.getFactors()) {
+                factor.setBaseSensitivityValue(factor.getEquationTerm().calculate(factorsState, factorGroup.getIndex()));
             }
         }
-        return new SensitivityValue(factor.getFactor(), value * PerUnit.SB, factor.getFunctionReference(), 0);
     }
 
     protected List<SensitivityValue> calculateSensitivityValues(List<SensitivityFactorGroup> factorGroups, DenseMatrix factorStates,
-                                                                DenseMatrix contingenciesStates, List<ComputedContingencyElement> contingencyElements) {
+                                                                DenseMatrix contingenciesStates, double[] flowStates, Collection<ComputedContingencyElement> contingencyElements) {
         List<SensitivityValue> sensitivityValuesContingencies = new ArrayList<>(factorGroups.stream().mapToInt(group -> group.getFactors().size()).sum());
+        setAlphasFlow(contingencyElements, flowStates, contingenciesStates);
         for (SensitivityFactorGroup factorGroup : factorGroups) {
-            setAlphas(contingencyElements, factorGroup, factorStates, contingenciesStates);
+            setAlphasSensi(contingencyElements, factorGroup, factorStates, contingenciesStates);
             for (LfSensitivityFactor factor : factorGroup.getFactors()) {
-                sensitivityValuesContingencies.add(createBranchSensitivityValue(factor, factorGroup, factorStates, contingenciesStates, contingencyElements));
+                sensitivityValuesContingencies.add(createBranchSensitivityValue(factor, contingenciesStates, contingencyElements));
             }
         }
         return sensitivityValuesContingencies;
     }
 
-    private void setAlphas(List<ComputedContingencyElement> contingencyElements, SensitivityFactorGroup sensitivityFactorGroup,
+    private void setAlphasFlow(Collection<ComputedContingencyElement> contingencyElements,
```

Comment:
Can you try to factorize this function with `setAlphaSensi` for maintainability?